src/Compilers/CSharp/Portable/Binder/LocalScopeBinder.cs

```diff
@@ -25,6 +26,35 @@ internal LocalScopeBinder(Binder next)
         internal LocalScopeBinder(Binder next, BinderFlags flags)
             : base(next, flags)
         {
+            var parentDepth = next.LocalScopeDepth;
+
+            if (parentDepth != Binder.TopLevelScope)
+            {
+                _localScopeDepth = parentDepth + 1;
+            }
+            else
+            {
+                //NOTE: TopLevel is special.
+                //For our purpose parameters and top level locals are on that level.
+                var parentScope = next;
+                while(parentScope != null)
+                {
+                    if (parentScope is InMethodBinder || parentScope is WithLambdaParametersBinder)
+                    {
+                        _localScopeDepth = Binder.TopLevelScope;
+                        break;
+                    }
+
+                    if (parentScope is LocalScopeBinder)
+                    {
+                        _localScopeDepth = Binder.TopLevelScope + 1;
+                        break;
+                    }
+
+                    parentScope = parentScope.Next;
+                    Debug.Assert(parentScope != null);
+                }
+            }
```

Comment:
Does this loop generate a different result than `_localScopeDepth = next.LocalScopeDepth + 1;`?

Reply:
the goal is to do exactly that , except in the case of parameter/top scopes that both are at level 1. We need this loop to detect such cases.

---
In reply to: [135185804](https://github.com/dotnet/roslyn/pull/21334#discussion_r135185804) [](ancestors = 135185804)